runner/run.go

```diff
@@ -165,8 +165,14 @@ func RunStage(runContext *TestRunContext, ftwCheck *check.FTWCheck, testCase sch
 		Protocol: testInput.GetProtocol(),
 	}
 
+	// Get the host header from the test input
+	host := testInput.GetHeaders().Get("Host")
+	if host == "" {
+		host = "localhost"
+	}
```

Comment:
@theseion Maybe we can add an url parse here, if we prefix the host with `http://`. E.g.
```golang
u, err := url.Parse("http://localhost%00")
	if err != nil {
		log.Fatal(err)
	}
```
For the test that fails:
```
2009/11/10 23:00:00 parse "http://localhost%00": invalid URL escape "%00"
```
And if `err != nil` we resort to just `localhost`?

Reply:
Let me have a chat with @airween. Let's see what he really needs.